engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/EmbeddedSubProcessTest.java

```diff
@@ -500,4 +523,44 @@ public void shouldTerminateSubProcessWithPendingParallelGateway() {
             tuple(BpmnElementType.SUB_PROCESS, ProcessInstanceIntent.ELEMENT_TERMINATED),
             tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_TERMINATED));
   }
+
+  @Test
+  public void shouldNotOverrideVariablesOnCompleteSubProcess() {
+    // given a process instance waiting for a job inside a sub process
+    ENGINE.deployment().withXmlResource(ONE_TASK_SUB_PROC_WITH_NON_INT_MSG_BNDRY_EVENT).deploy();
```

Comment:
`ONE_TASK_SUB_PROC_WITH_NON_INT_MSG_BNDRY_EVENT` this is a long name for a variable :sweat_smile: 

(optional) We could write the BPMN directly in the test case instead of using a constant. It could make the test case easier to understand if the BPMN is closer to the code that tests the flow. Otherwise, you may need to jump from bottom to top and back.